### PR TITLE
feat: new deposit status route

### PR DESCRIPTION
**Pull request by dohaki**
Closes ACX-1628

Requires #889 and #892

This PR implements the redesigned deposit status as a new route to provide better UX. The diff got blown up a bit by that because a lot of boilerplate code had to be added etc. I outlined the main changes with some comments.

### CODE REVIEW

File: src/hooks/useLocalPendingDeposits.ts
Comment (dohaki) [inline src/hooks/useLocalPendingDeposits.ts:1]: This was refactored to `src/utils/local-deposits`. The reasoning behind this:
- There was no need for these helpers to be inside a hook
- Allows us to use these outside of react components

File: src/views/Bridge/hooks/useBridge.ts
Comment (dohaki) [inline src/views/Bridge/hooks/useBridge.ts:1]: Because the deposit status tracking is now a separate route, the logic in this hook got a lot more cleaner and easier to reason about.

File: src/views/DepositStatus/components/DepositStatusLowerCard.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  fromBridgePagePayload?: FromBridgePagePayload;
+};
+
+export function DepositStatusLowerCard({
+  fromChainId,
+  toChainId,
+  bridgeTokenSymbol,
+  fromBridgePagePayload,
+}: Props) {
+  const { quote } = fromBridgePagePayload || {};
```
Comment (dohaki) [inline src/views/DepositStatus/components/DepositStatusLowerCard.tsx:27]: With the current implementation, this component will only display the fees for the respective user that did the transfer. In the future, we could add some fallback logic here that tries to fetch data from the Scraper or Contract. So this could also serve as a generic status page that can be visited by anyone.

File: src/views/DepositStatus/utils.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    status: "pending" as const,
+    filled: "0",
+    sourceChainId: sendDepositArgs.fromChain,
+    destinationChainId: sendDepositArgs.toChain,
+    assetAddr: sendDepositArgs.tokenAddress,
+    depositorAddr: utils.getAddress(fromBridgePagePayload.account),
+    recipientAddr: sendDepositArgs.toAddress,
+    message: sendDepositArgs.message || "0x",
+    amount: BigNumber.from(sendDepositArgs.amount).toString(),
+    depositTxHash: data.depositTxReceipt.transactionHash || "0x",
```
Comment (james-a-morris) [inline src/views/DepositStatus/utils.ts:25]: Should we throw if the deposit hash doesn't exist?

File: src/views/DepositStatus/utils.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+export function convertForFillQuery(
+  data: Awaited<ReturnType<typeof getFillByDepositTxHash>>,
+  fromBridgePagePayload: FromBridgePagePayload
+) {
+  const { sendDepositArgs, quote } = fromBridgePagePayload;
+
+  return {
+    depositId: Number(
+      data.depositByTxHash.parsedDepositLog.args.depositId || 0
```
Comment (james-a-morris) [inline src/views/DepositStatus/utils.ts:58]: Same here ^

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+`;
+
+const AnimatedDividerToChain = styled(AnimatedDivider)<{
+  status: DepositStatus;
+}>`
+  background: ${({ status }) =>
+    status === "depositing"
+      ? "#9daab3"
+      : status === "filling"
+      ? "#FFFFFF"
```
Comment (james-a-morris) [inline src/views/DepositStatus/components/DepositStatusUpperCard.tsx:263]:
```suggestion
      ? "#ffffff"
```

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+  const { elapsedSeconds: fillTxElapsedSeconds } = useElapsedSeconds(
+    depositTxCompletedTime,
+    fillTxCompletedTime
+  );
+
+  const status = !depositTxCompletedTime
+    ? "depositing"
+    : !fillTxCompletedTime
+    ? "filling"
+    : "filled";
```
Comment (james-a-morris) [inline src/views/DepositStatus/components/DepositStatusUpperCard.tsx:78]: Thoughts on codifying this as an `enum` or `type BridgeStatus = "depositing" | "filling" | "filled"`

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+  [ChainId.MAINNET]: <EthereumGrayscaleLogo />,
+  [ChainId.ZK_SYNC]: <ZkSyncGrayscaleLogo />,
+  [ChainId.BASE]: <BaseGrayscaleLogo />,
+  // testnets
+  [ChainId.GOERLI]: <EthereumGrayscaleLogo />,
+  [ChainId.ARBITRUM_GOERLI]: <EthereumGrayscaleLogo />,
+  [ChainId.MUMBAI]: <EthereumGrayscaleLogo />,
+  [ChainId.ZK_SYNC_GOERLI]: <ZkSyncGrayscaleLogo />,
+  [ChainId.BASE_GOERLI]: <BaseGrayscaleLogo />,
+};
```
Comment (james-a-morris) [inline src/views/DepositStatus/components/DepositStatusUpperCard.tsx:39]: Should we place this in constants? This feels like a more general structure

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+import { ChainId, QUERIESV2 } from "utils";
+
+import { useElapsedSeconds } from "../hooks/useElapsedSeconds";
+import { useDepositTracking } from "../hooks/useDepositTracking";
+import { DepositTimesCard } from "./DepositTimesCard";
+import { ElapsedTime } from "./ElapsedTime";
+import { DepositStatus, FromBridgePagePayload } from "../types";
+
+const grayscaleLogos: {
+  [key: number]: JSX.Element;
```
Comment (james-a-morris) [inline src/views/DepositStatus/components/DepositStatusUpperCard.tsx:25]: Should we use `Record< >` or keep the index format for consistency? I've always thought `Record<>` is a good tradeoff, and it is more succinct - but I'm not sure the benefits of this type

File: src/views/DepositStatus/DepositStatus.tsx
Comment (james-a-morris) [inline src/views/DepositStatus/DepositStatus.tsx:1]: Do we handle the case where a tx hash is passed that is not an across related transaction?

File: src/views/Bridge/hooks/useBridgeAction.ts [hunk 8]
```diff
@@ -108,76 +97,30 @@ export function useBridgeAction(
...
+      {
+        fromBridgePagePayload: {
+          sendDepositArgs: frozenPayload,
+          quote: frozenQuote,
+          referrer,
+          account: frozenAccount,
+          timeSigned: Date.now(),
+          tokenPrice,
+        },
       }
```
Comment (james-a-morris) [inline src/views/Bridge/hooks/useBridgeAction.ts:123]: Is there a cleaner way to do this? Embedding the class constructor in the string formatter was a tad hard to follow.

**Review by james-a-morris — COMMENTED** [review]
Left a few comments

File: src/views/DepositStatus/utils.ts (hunk 2 above)
Comment (dohaki) [reply src/views/DepositStatus/utils.ts:25]: We can actually remove this because this was initially a workaround for some type issues that I fixed already.

File: src/views/DepositStatus/utils.ts (hunk 3 above)
Comment (dohaki) [reply src/views/DepositStatus/utils.ts:58]: We can actually remove this because this was initially a workaround for some type issues that I fixed already.

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 5 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx:78]: So there is already a type `src/views/DepositStatus/types.ts`:
```
export type DepositStatus = "depositing" | "filling" | "filled";
```

And I think it suffices here in this case that the type gets inferred correctly.

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 4 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx:263]: I think using the COLORS constants makes sense here. WIll change

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 6 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx:39]: Hmm not sure... This would require us to change the file extension of the `constants.ts` file to `constants.tsx`. And currently these logos are only used here

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 7 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx:25]: Yea agree! I copy-pasted a lot from the old components and the index format has been used there. Will change

File: src/views/DepositStatus/DepositStatus.tsx
Comment (dohaki) [reply src/views/DepositStatus/DepositStatus.tsx:1]: Not explicitly... It would just show a never-ending loading screen

File: src/views/Bridge/hooks/useBridgeAction.ts (hunk 8 above)
Comment (dohaki) [reply src/views/Bridge/hooks/useBridgeAction.ts:123]: I could pull it out of the templated string maybe?

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 4 above)
Comment (dohaki) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx]: https://github.com/across-protocol/frontend-v2/pull/893/commits/95cb1ba1940d0e32f3510ec77e0893ff7ff7ef0f

File: src/views/DepositStatus/utils.ts (hunk 3 above)
Comment (dohaki) [reply src/views/DepositStatus/utils.ts]: https://github.com/across-protocol/frontend-v2/pull/893/commits/84e229edea6e1319393d5f4e4bdad800e510da67

File: src/views/DepositStatus/utils.ts (hunk 2 above)
Comment (dohaki) [reply src/views/DepositStatus/utils.ts]: https://github.com/across-protocol/frontend-v2/pull/893/commits/84e229edea6e1319393d5f4e4bdad800e510da67

File: src/views/Bridge/hooks/useBridgeAction.ts (hunk 8 above)
Comment (dohaki) [reply src/views/Bridge/hooks/useBridgeAction.ts]: https://github.com/across-protocol/frontend-v2/pull/893/commits/17901501c0936979b6325e8901891c193bd7c7f9

File: src/views/DepositStatus/components/DepositStatusUpperCard.tsx (hunk 6 above)
Comment (james-a-morris) [reply src/views/DepositStatus/components/DepositStatusUpperCard.tsx]: Ah shoot good point with the need for a file change.

File: src/views/DepositStatus/DepositStatus.tsx
Comment (james-a-morris) [reply src/views/DepositStatus/DepositStatus.tsx:1]: Could we maybe check that the txn is related to the spoke pool contract?

**Review by james-a-morris — COMMENTED** [review]
One more question: https://github.com/across-protocol/frontend-v2/pull/893#discussion_r1378864556

File: src/views/DepositStatus/DepositStatus.tsx
Comment (dohaki) [reply src/views/DepositStatus/DepositStatus.tsx:1]: So I added some handling of invalid deposit tx hash here https://github.com/across-protocol/frontend-v2/pull/893/commits/36d753c0f052e05f9e17af62d372589940969798. This covers the cases:
- Invalid tx hash format
- Valid tx hash format and mined but not from a SpokePool

One case that this does not cover is:
- Valid tx hash format but not mined yet

I am not sure how we can check whether a not mined tx is from a SpokePool or not.

**Review by james-a-morris — APPROVED** [review]
The design matches the spec + all outstanding questions have been answered. LGTM
